wallet/controllers.go

```diff
@@ -0,0 +1,199 @@
+package wallet
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/hex"
+	"errors"
+	"net/http"
+	"os"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/brave-intl/bat-go/middleware"
+	"github.com/brave-intl/bat-go/utils/altcurrency"
+	errorutils "github.com/brave-intl/bat-go/utils/errors"
+	"github.com/brave-intl/bat-go/utils/handlers"
+	"github.com/brave-intl/bat-go/utils/httpsignature"
+	"github.com/brave-intl/bat-go/utils/requestutils"
+	walletutils "github.com/brave-intl/bat-go/utils/wallet"
+	"github.com/brave-intl/bat-go/utils/wallet/provider/uphold"
+	"github.com/go-chi/chi"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Router for suggestions endpoints
+func Router(service *Service) chi.Router {
+	r := chi.NewRouter()
+	r.Method("POST", "/{paymentId}/claim", middleware.HTTPSignedOnly(service)(middleware.InstrumentHandler("LinkWalletCompat", LinkWalletCompat(service))))
+	r.Method("GET", "/{paymentId}", middleware.InstrumentHandler("GetWallet", GetWallet(service)))
+	r.Method("POST", "/", middleware.HTTPSignedOnly(service)(middleware.InstrumentHandler("PostCreateWallet", PostCreateWallet(service))))
+	return r
+}
+
+// LookupPublicKey based on the HTTP signing keyID, which in our case is the walletID
+func (service *Service) LookupPublicKey(ctx context.Context, keyID string) (*httpsignature.Verifier, error) {
+	var publicKey httpsignature.Ed25519PubKey
+	// hex encoded public key
+	publicKey, err := hex.DecodeString(keyID)
+	if err != nil {
+		return nil, err
+	}
+	tmp := httpsignature.Verifier(publicKey)
```

Comment:
Is `Verifier` a function? I can't find it in the library, I see that it's an [interface](https://github.com/brave-intl/bat-go/blob/49ac91e3d9103493025b0faa7b47fee16c3ef346/utils/httpsignature/httpsignature.go)

Reply:
yeah, we're just casting it as an interface. so that it meets the requirements of `HTTPSignedOnly` in `middleware/http_signed`